x/compute/client/cli/tx.go

```diff
@@ -135,16 +139,22 @@ func InstantiateContractCmd(cdc *codec.Codec) *cobra.Command {
 			if label == "" {
 				return fmt.Errorf("label is required on all contracts")
 			}
+
 			route := fmt.Sprintf("custom/%s/%s/%s", types.QuerierRoute, keeper.QueryContractAddress, label)
 			res, _, err := cliCtx.Query(route)
 			if res != nil {
 				return fmt.Errorf("label already exists. You must choose a unique label for your contract instance")
 			}
 
-			wasmCtx := wasmUtils.WASMContext{CLIContext: cliCtx}
+			initMsg.CodeHash, err = GetCodeHashByCodeId(cliCtx, args[0])
+			if err != nil {
+				return err
+			}
 
-			initMsg := []byte(args[1])
-			initMsg, err = wasmCtx.Encrypt(initMsg)
+			// todo: Add check that this is valid json and stuff
+			initMsg.Msg = []byte(args[1])
+
```

Comment:
You don't really need to check that it's a JSON, since the format is defined in the contract

Reply:
You don't _need_ to, but if you make a mistake it gives you a weird error instead of telling you exactly what you did wrong

Reply:
still, if you add this though, you'll lock down contract from using other formats if they choose to use something different than the current `cosmwasm-std`

Reply:
It can come with a flag that turns the check off.. it's just a CLI thing, and I'd rather people have an easier time to use what we have now than have an easier time supporting some advanced feature that we don't know anyone will want

Reply:
I suppose that's fair. if we need to we can loosed that in the future. The messages that the contract returns are required to be JSON anyways, so it's unlikely people will do anything different here.